src/miner.cpp

```diff
@@ -380,19 +375,17 @@ bool static ScanHash(const CBlockHeader *pblock, uint32_t& nNonce, uint256 *phas
     hasher.Write((unsigned char*)&ss[0], 76);
 
     while (true) {
-        nNonce++;
+        pblock->nNonce++;
 
         // Write the last 4 bytes of the block header (the nonce) to a copy of
         // the double-SHA256 state, and compute the result.
-        CHash256(hasher).Write((unsigned char*)&nNonce, 4).Finalize((unsigned char*)phash);
+        CHash256(hasher).Write((unsigned char*)&pblock->nNonce, 4).Finalize((unsigned char*)&hash);
 
-        // Return the nonce if the hash has at least some zero bits,
-        // caller will check if it has enough to reach the target
-        if (((uint16_t*)phash)[15] == 0)
+        if (CheckProofOfWork(hash, pblock->nBits, chainparams))
```

Comment:
Does it give a lot of overhead to call CheckProofOfWork for every nonce, instead of the 'some zero bits' check?